packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_dex_poolswap.test.ts

```diff
@@ -126,7 +110,90 @@ describe('DFI to DOG', () => {
     expect(prevouts[0].value.toNumber()).toBeLessThan(10)
     expect(prevouts[0].value.toNumber()).toBeGreaterThan(9.999)
   })
-})
 
-// TODO(jellyfish): test alternative poolpair
-// TODO(jellyfish): test alternative maxPrice
+  it('should poolSwap with maxPrice', async () => {
+    await providers.randomizeEllipticPair()
+    await container.waitForWalletBalanceGTE(1)
+    const addressLP = await container.getNewAddress()
+    await addPoolLiquidity(container, {
+      tokenA: 'DFI',
+      amountA: 100,
+      tokenB: 'PIG',
+      amountB: 20,
+      shareAddress: addressLP
+    })
+    await providers.setupMocks()
+    await utxosToAccount(container, 100, { address: await providers.getAddress() })
+    await sendTokensToAddress(container, await providers.getAddress(), 2, 'PIG')
+    await fundEllipticPair(container, providers.ellipticPair, 10)
+    const script = await providers.elliptic.script()
+
+    const txn = await builder.dex.poolSwap({
+      fromScript: script,
+      fromTokenId: pairs.PIG.tokenA,
+      fromAmount: new BigNumber('2'),
+      toScript: script,
+      toTokenId: pairs.PIG.tokenB,
+      maxPrice: {
+        integer: new BigNumber('5'),
+        fraction: new BigNumber('0')
+      }
+    }, script)
+
+    const outs = await sendTransaction(container, txn)
+    expect(outs[0].value).toStrictEqual(0)
+    expect(outs[1].value).toBeLessThan(10)
+    expect(outs[1].value).toBeGreaterThan(9.999)
+    expect(outs[1].scriptPubKey.addresses[0]).toStrictEqual(await providers.getAddress())
+
+    const account = await jsonRpc.account.getAccount(await providers.getAddress())
+    expect(account).toContain('98.00000000@DFI')
+    expect(account).toContain('2.39215687@PIG')
+
+    const poolPair = await jsonRpc.poolpair.getPoolPair('DFI-PIG', true)
+    const pair = Object.values(poolPair)[0]
+
+    expect(pair.totalLiquidity.toFixed(8)).toStrictEqual('44.72135954')
+    expect(pair.reserveA.toFixed(8)).toStrictEqual('102.00000000')
+    expect(pair.reserveB.toFixed(8)).toStrictEqual('19.60784313')
+
+    const prevouts = await providers.prevout.all()
+    expect(prevouts.length).toStrictEqual(1)
+    expect(prevouts[0].value.toNumber()).toBeLessThan(10)
+    expect(prevouts[0].value.toNumber()).toBeGreaterThan(9.999)
+  })
+
+  it('should fail poolSwap due to maxPrice lower than reserveA/reserveB', async () => {
+    await providers.randomizeEllipticPair()
+    await container.waitForWalletBalanceGTE(1)
+    const addressLP = await container.getNewAddress()
+    await addPoolLiquidity(container, {
+      tokenA: 'DFI',
+      amountA: 100,
+      tokenB: 'PIG',
```

Comment:
There are 2 tests that use `DFI-PIG`

should use a new pool for this test, since `should poolSwap with maxPrice` would have influenced the price giving you a side effects.